dds/DCPS/transport/framework/TransportReassembly.cpp

```diff
@@ -373,7 +384,23 @@ TransportReassembly::data_unavailable(const SequenceNumber& dataSampleSeq,
                                       const RepoId& pub_id)
 {
   const FragKey key(pub_id, dataSampleSeq);
-  fragments_.erase(key);
+  erase_i(fragments_.find(key));
+}
+
+void
+TransportReassembly::erase_i(FragInfoMap::iterator pos)
+{
+  if (pos != fragments_.end()) {
+    const FragKey& key = pos->first;
+    ExpirationQueue::iterator pos2 = expiration_queue_.find(pos->second.expiration_);
```

Comment:
I think lower_limit() and upper_limit() would help here?

Reply:
With multimap, find() is not guaranteed to give us the first match.  I'll change on my branch to using equal_range.